internal/bundle/plain_test.go

```diff
@@ -0,0 +1,124 @@
+package bundle
+
+import (
+	"io/fs"
+	"testing/fstest"
+
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+)
+
+var _ = Describe("Plain bundle v0 loader", func() {
+	It("should fail if there is no manifests directory", func() {
+		mockfs := fstest.MapFS{}
+		_, err := LoadPlainV0(mockfs)
+		Expect(err.Error()).To(ContainSubstring("file does not exist"))
+	})
+	It("should fail if there is an empty manifests directory", func() {
+		mockfs := fstest.MapFS{
+			"manifests": {
+				Mode: fs.ModeDir,
+			},
+		}
+		_, err := LoadPlainV0(mockfs)
+		Expect(err.Error()).To(ContainSubstring("found zero files"))
+	})
+	It("should fail if there is a file, but no yaml objects", func() {
+		mockfs := fstest.MapFS{
+			"manifests/invalid.yaml": {
+				Data: []byte(""),
+			},
+		}
+		_, err := LoadPlainV0(mockfs)
+		Expect(err.Error()).To(ContainSubstring("required to contain at least one object"))
+	})
+	It("should fail if any of the files are invalid json or yaml", func() {
+		mockfs := fstest.MapFS{
+			"manifests/valid.yaml": {
+				Data: []byte("kind: ofnecessary"),
+			},
+			"manifests/invalid.yaml": {
+				Data: []byte("not;valid"),
+			},
+		}
+		_, err := LoadPlainV0(mockfs)
+		Expect(err.Error()).To(ContainSubstring("cannot unmarshal string into Go value"))
+	})
+	It("should fail if the YAML or JSON objects do not specify `kind`", func() {
+		mockfs := fstest.MapFS{
+			"manifests/invalid.yaml": {
+				Data: []byte("almost: valid"),
+			},
+		}
+		_, err := LoadPlainV0(mockfs)
+		Expect(err.Error()).To(ContainSubstring("Object 'Kind' is missing"))
+	})
+	It("should pass if the YAML is valid and contains a `kind` element", func() {
+		mockfs := fstest.MapFS{
+			"manifests/valid.yaml": {
+				Data: []byte("kind: ofnecessary"),
+			},
+		}
+		_, err := LoadPlainV0(mockfs)
+		Expect(err).To(BeNil())
+	})
+	It("should pass if the JSON is valid and contains a `kind` element", func() {
+		mockfs := fstest.MapFS{
+			"manifests/valid-json.yaml": {
+				Data: []byte(`{"kind": "ofnecessary"}`),
+			},
+			"manifests/valid.yaml": {
+				Data: []byte("kind: ofnecessary"),
+			},
```

Comment:
Seems like this test is identical to the one below. Looks like this is supposed to test a JSON file by itself.
```suggestion
			"manifests/valid.json": {
				Data: []byte(`{"kind": "ofnecessary"}`),
			},
```